controllers/che/checluster_controller.go

```diff
@@ -260,6 +260,29 @@ func (r *CheClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 		CheCluster: instance,
 	}
 
+	if isPreviousVersionDeployed(instance) {
+		// Current operator is newer than deployed Che
+		backupCR, err := getBackupCR(deployContext)
+		if err != nil {
```

Comment:
I would like to see something  `isBackupInProgress`

Reply:
It is better to do analysis like it is already implemented. This way we will not query for backupCR twice.